f5_openstack_agent/lbaasv2/drivers/bigip/exceptions.py

```diff
@@ -356,3 +364,79 @@ class VXLANDeleteException(F5AgentException):
 
 class BigIPNotLicensedForVcmp(F5AgentException):
     pass
+
+
+class F5MissingDependencies(F5AgentException):
+    default_msg = "%s cannot start due to missing dependency" % \
+        str(sys.argv[0])
+    message_format = "(%d) %s: %s [%s:%s]"
+    default_errno = errno.ENOSYS
+    default_project = 'neutron'
+    default_name = 'f5-oslbaasv2-agent'
+
+    def __init__(self, *args, **kargs):
+        self.__set_message(args, kargs)
+        super(F5MissingDependencies, self).__init__(self.message)
+        self.__logger()
+        self.__log_error()
+        self.__check_debug()
+
+    def __check_debug(self):
+        print("__check_debug")
+        debug = False
+        for item in sys.argv:
+            if 'f5-openstack-agent.ini' in item:
+                with open(item, 'r') as fh:
+                    line = fh.readline()
+                    print('line', line)
+                    debug_re = \
+                        re.compile('debug\s*=\s*([Tt]rue|[Ff]alse|[01])')
+                    while line:
+                        match = debug_re.search(line)
+                        if match:  # there should only be one!
+                            if re.search('[tT]|1', match.group(1)):
+                                debug = True
+                            break
+                        line = fh.readline()
+        if debug:
+            # our sys.exc_info queue should still be pointing at our ImportE
+            traceback.print_exc()
+
+    def __logger(self):
+        try:
+            self._logger = \
+                logging.getLogger('%s:%s' %
+                                  (self.default_project, self.default_name))
```

Comment:
Can we set the `self.__logger` name to be the `__name__` of the `raising` module?  Perhaps by extracting that information from `inspect`?